core/capabilities/ccip/ccipaptos/addresscodec_test.go

```diff
@@ -0,0 +1,121 @@
+package ccipaptos
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func generateAddressBytes32() [32]byte {
+	var result [32]byte
+	_, err := rand.Read(result[:])
+	if err != nil {
+		panic(fmt.Sprintf("failed to generate random address bytes: %v", err))
+	}
+	return result
+}
+
+func generateAddressBytes() []byte {
+	a := generateAddressBytes32()
+	return a[:]
+}
+
+func generateAddressString() string {
+	addressBytes := generateAddressBytes()
+	addressString, err := addressBytesToString(addressBytes)
+	if err != nil {
+		panic(fmt.Sprintf("failed to generate random address string: %v", err))
+	}
+	return addressString
+}
+
+func TestAddressBytesToString(t *testing.T) {
+	tests := []struct {
+		name     string
+		inHex    string
+		isErr    bool
+		expected string
+	}{
+		{
+			"empty",
+			"",
+			true,
+			"",
+		},
+		{
+			"equal to 32 bytes",
+			"0102030405060102030405060102030405060102030405060102030405060101",
+			false,
+			"0x0102030405060102030405060102030405060102030405060102030405060101",
+		},
+		{
+			"longer than required",
+			"0102030405060102030405060102030405060102030405060102030405060101FFFFFFFFFF",
+			true,
+			"",
+		},
```

Comment:
Can we have a test case for `shorter than required`? In order to go through the padding logic.